_posts/2020-05-30-Metal3-Cluster-Deployment.md

```diff
@@ -0,0 +1,354 @@
+- Finally a yaml file is generated with all the details for the cluster/controlplane/worker, which is then passed to K8s module in ansible which creates the resources.
+- There are also manifests stored in ${Manifests} directory - [`${metal3-dev-env}/vm-setup/roles/v1aX_integration_test/files/manifests`](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test), for the cluster which can be used to deprovision the cluster at a later point in time.
+> **Note** : * the `manifests` directory is not present but created when you first use the scripts.*
+- Centos or Ubuntu images [are downloaded](https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/v1aX_integration_test/tasks/download_image.yml) when provisioning a controlplane or a worker  
+
+<br/>
+<br/>
+
+
+!["An Overview of various resources generated while provisioning and their relationship amongst themselves"](/assets/2020-05-30-Metal3-Cluster-Deployment/metal3-bmetal-arch-overview.png)  
+<br/>
+<br/>
+
+
+### Provision Cluster
+This script, located at the path - `${metal3-dev-env}/scripts/v1alphaX/provision_clusters.sh`, provisions the cluster by creating a `Metal3Cluster` resource. 
+
+To see if you have a successful Cluster resource creation( the cluster still doesn't have a Controlplane or Workers ), just do - 
+```console
+kubectl get Metal3Cluster ${CLUSTER_NAME} -n metal3
+```
+This will return the cluster deployed, and you can check the cluster details by describing the returned resource.
+
+Here is what a `Cluster` resource looks like :
+```console
+kubectl describe Cluster ${CLUSTER_NAME} -n metal3
+```
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: Cluster
+metadata:
+  [......]
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+      - 192.168.0.0/18
+    services:
+      cidrBlocks:
+      - 10.96.0.0/12
+  controlPlaneEndpoint:
+    host: 192.168.111.249
+    port: 6443
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+    kind: KubeadmControlPlane
+    name: eko
+    namespace: metal3
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    kind: Metal3Cluster
+    name: eko
+    namespace: metal3
+status:
+  infrastructureReady: true
+  phase: Provisioned
+```
+  
+<br/>
+<br/>
+
+### Provision Controlplane
+
+This script, located at the path - ${metal3-dev-env}/scripts/v1alphaX/provision_clusters.sh, provisions the controlplane or the master of the cluster. As part of the controlplane provisioning, the generated manifests are for cluster, controlplane and machinedeployment. The **MachineDeployment** creates a **Machine** which picks up a BareMetalHost satisfying its requirements as the controlplane node, and it is then provisioned by the BareMetalOperator. there are other resources created in the process of provisioning, like `KubeadmControlPlane` and `Metal3MachineTemplate` resource which are involved in the provisioning process.
```

Comment:
use `back ticks` for the script path.

Also, usage of **bold** looks inconsistent... why in **MachineDeployment** and **Machine** but not on BareMetalHost?  Maybe use back ticks for these three as well, as they refer to resource types (`MachineDeployment`, `Machine`, `BareMetalHost` - like you have later in the paragraph for `KubeadmControlPlane` and `Metal3MachineTemplate`.

I didn't double check the consistency of this type of markup across the post, but it's worth checking